quattor/functions/validation.pan

```diff
@@ -139,3 +139,11 @@ function valid_absolute_file_paths = {
     };
     true;
 };
+
+@documentation{
+    desc = This types is meant to be used to check if strings provided are \
+        valid user names. This is however designed to be overriden by \
+	site-specific policies. The regex used here as default is the \
+	one suggested by the useradd man page.
+}
+type valid_user = string with match(SELF, '^[a-zA-Z_][a-zA-Z0-9_-]{0,30}([a-zA-Z0-9_-]|\$)?$');
```

Comment:
The problem with this is that validation checks can only be made more-specific in later files, so this would mean that it's impossible for any site to have a regex that's less restrictive (e.g. length >30, or including a .)

```suggestion
variable VALID_USER_PATTERN ?= '^[a-zA-Z_][a-zA-Z0-9_-]{0,30}([a-zA-Z0-9_-]|\$)?$';
type valid_user = string with match(SELF, VALID_USER_PATTERN);
```